File: gestiondestock/src/main/java/fr/esic/gestiondestock/services/ArticleService.java
package fr.esic.gestiondestock.services;

import fr.esic.gestiondestock.dto.ArticleDto;
import fr.esic.gestiondestock.dto.LigneCommandeClientDto;
import fr.esic.gestiondestock.dto.LigneCommandeFournisseurDto;
import fr.esic.gestiondestock.dto.LigneVenteDto;

import java.util.List;

public interface ArticleService {

    ArticleDto save(ArticleDto dto);

    ArticleDto findById(Integer id);

    ArticleDto findByCodeArticle(String codeArticle);

    List<ArticleDto> findAll();
    
    List<LigneVenteDto> findHistoriqueVente(Integer idArticle);
    
    List<LigneCommandeClientDto> findHistoriqueCommandeClient(Integer idArticle);
    
    List<LigneCommandeFournisseurDto> findHistoriqueCommandeFournisseur(Integer idArticle);
    
    List<ArticleDto> findAllArticleByIdCategorie(Integer idCategory);

    void delete(Integer id);
}

File: frontend-gestion-de-stock/src/app/services/article/article.service.ts
import { Injectable } from '@angular/core';
import { Observable, of } from 'rxjs';
import { ArticleDto } from 'src/gs-api/src/models';
import { ArticlesService } from 'src/gs-api/src/services';
import { UserService } from '../user/user.service';

@Injectable({
  providedIn: 'root'
})
export class ArticleService {

  constructor(
    private userService: UserService,
    private articleService: ArticlesService
  ) { }

  enregistrerArticle(articleDto: ArticleDto): Observable<ArticleDto> {
    articleDto.idEntreprise = this.userService.getConnectedUser().entreprise?.id;
    return this.articleService.save(articleDto);
  }

  findAllArticles(): Observable<ArticleDto[]> {
    return this.articleService.findAll();
  }

  findArticleById(idArticle?: number): Observable<ArticleDto> {
    console.log(idArticle);
    if (idArticle) {
      return this.articleService.findById(idArticle);
    }
    return of();
  } 

  deleteArticle(idArticle?: number): Observable<any> {
    if (idArticle) {
      return this.articleService.delete(idArticle);
    }
    return of();
  }

  findArticleByCode(codeArticle: string): Observable<ArticleDto>  {
    return this.articleService.findByCodeArticle(codeArticle);
  }
}
